src/libraries/System.Private.CoreLib/src/System/Globalization/CompareInfo.Icu.cs

```diff
@@ -411,32 +411,38 @@ private unsafe int IndexOfOrdinalHelper(ReadOnlySpan<char> source, ReadOnlySpan<
             }
         }
 
-        private unsafe bool IcuStartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options)
+        // this method sets '*matchLengthPtr' (if not nullptr) on exit
+        private unsafe bool IcuStartsWith(ReadOnlySpan<char> source, ReadOnlySpan<char> prefix, CompareOptions options, int* matchLengthPtr)
```

Comment:
Yeah, there's a little bit of inconsistency here. I ended up using `matchLengthPtr` because it's what existing methods like `NlsIndexOfCore` and `IcuIndexOfCore` use. In other pinvoke-related code paths in the runtime we tend to use the C-style `p*` and `dw*` prefixes, but that never consistently caught on in `CompareInfo` for some reason.